src/planscape/planning/views.py

```diff
@@ -266,33 +326,51 @@ def get_planning_area_by_id(request: HttpRequest) -> HttpResponse:
     try:
         user = request.user
         if not user.is_authenticated:
-            return JsonResponse({"error": "Authentication Required"}, status=401)
+            return Response(
+                {"error": "Authentication Required"},
+                status=status.HTTP_401_UNAUTHORIZED,
+            )
 
         if "id" not in request.GET:
-            return JsonResponse(
-                {"error": "Missing required parameter 'id'"}, status=400
+            return Response(
+                {"error": "Missing required parameter 'id'"},
+                status=status.HTTP_400_BAD_REQUEST,
             )
-        return JsonResponse(
-            _serialize_planning_area(
-                get_object_or_404(
-                    user.planning_areas.annotate(
-                        scenario_count=Count("scenarios", distinct=True)
-                    ).annotate(scenario_latest_updated_at=Max("scenarios__updated_at")),
-                    id=request.GET["id"],
-                ),
-                True,
-                context={"request": request},
+
+        planning_area = (
+            PlanningArea.objects.filter(id=request.GET["id"])
+            .annotate(scenario_count=Count("scenarios", distinct=True))
+            .annotate(
+                scenario_latest_updated_at=Coalesce(
+                    Max("scenarios__updated_at"), "updated_at"
+                )
             )
+            .first()
+        )
+
+        if not planning_area:
+            return Response(
+                {"message": "Planning area not found with this ID"},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        if not PlanningAreaPermission.can_view(user, planning_area):
+            return Response(
+                {"message": "User has no access to this planning area."},
+                status=status.HTTP_403_FORBIDDEN,
+            )
```

Comment:
Currently, we return a 404 if someone doesn't own the planning area, because we were doing a get_object_or_404. Here we have the opportunity to return a 403 in the case where they have no permissions.
I don't see the benefit of security-by-obscurity in returning a 404, but I'm open to changing this back.

Reply:
i like it, I think the benefit outweighs the possible very-paranoid scenario of someone knowing that a planning area exists, for someone else than them.